data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -373,9 +451,16 @@ class ApiServerClient {
     assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to delete resource`);
     assert.ok(opts.resourceType, `Property 'resourceType' is required to delete resource`);
     assert.ok(opts.resourceId, `Property 'resourceId' is required to delete resource`);
+    const namespaceId = opts.namespaceId ? opts.namespaceId : this.getNamespaceId(opts.resourceId);
     return Promise.try(() => this.init())
       .then(() => apiserver.apis[opts.resourceGroup][CONST.APISERVER.API_VERSION]
-        .namespaces(CONST.APISERVER.NAMESPACE)[opts.resourceType](opts.resourceId).delete())
+        .namespaces(namespaceId)[opts.resourceType](opts.resourceId).delete())
+      .then((res) => {
+        if (namespaceId !== CONST.APISERVER.DEFAULT_NAMESPACE && opts.resourceType === CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEINSTANCES) {
```

Comment:
i guess we clean up namespace only when instance is being deleted. 
Just thought of it,  do we even delete binding and other resources related to instance upon instance deletion as a cascading operation? or they are just left as is like zombies?

Reply:
bindings are deleted as part of binding deletion. Also, you cant delte namespace without deleting the resources within it. Hence the binding and secrets have to be deleted.